packages/ui/src/theme/css/component/checkbox.scss

```diff
@@ -2,8 +2,11 @@
   cursor: var(--amplify-components-checkbox-cursor);
   align-items: var(--amplify-components-checkbox-align-items);
   gap: inherit;
-  &[data-disabled='true'] {
-    cursor: var(--amplify-components-checkbox-disabled-cursor);
+  --amplify-internal-checkbox-disabled-cursor: var(
+    --amplify-components-checkbox-disabled-cursor
+  );
+  &--disabled {
+    cursor: var(--amplify-internal-checkbox-disabled-cursor);
```

Comment:
I think we do not need `--amplify-internal-checkbox-disabled-cursor` right? I don't see we need to override this anywhere